### PR TITLE
Login Backend Model and Endpoint

### LINKED ISSUE #8: [DEV] Create a user login endpoint
**Related To**
- [User Login](https://github.com/tysonkaufmann/su-go/issues/6)

**Description**
Create an API endpoint `/api/user/login` which will serve `POST` requests with a body that looks like: **(These are subject to change)**

```
{
  "username":"",
  "password":""
}
```
On Success, the API endpoint should return  e.g **(These are subject to change)**

```
{
  "status":"200",
  "success":"true",
  "token":"",
  "expiresin":"",
  "msg":"Login successful"
}
```
On Error, the API endpoint should return e.g **(These are subject to change)**

```
{
  "status":"200",
  "success":"false",
  "msg":"Invalid username/password"
}
```

**Development Steps**

- Create integration tests to fully test the API endpoint `/api/user/login` (TDD)
- Route the endpoint in `backend/routes/userAuth`
- Create unit tests for any code added in `backend/controllers/userAuth` (TDD)
- Add code to the `backend/controllers/userAuth` to serve the API request

### LINKED ISSUE #9: [DEV] Create an Auth collection and UserAuth model
**Related To**
- [User Login](https://github.com/tysonkaufmann/su-go/issues/6)

**Description**
Because the data will be stored in a NoSQL database we will create a collection and a model where the schema would look like,
```
STRING username
STRING password
```

**Development Steps**
- Create a collection in MongoDB called `Auth`
- Add unit tests for functions in the model (TDD)
- Create a UserAuth model in `backend/models/userAuth`

### PR DESCRIPTION
**Pull request by Mitul264**
# Description

Created an API endpoint `/api/user/login` which will serve `POST` requests with a body that looks like: 

```
{
  "username":"",
  "password":""
}
```
# Closes #8 

Created a collection and a model where the schema would look like,
```
STRING username
STRING password
```
# Closes #9 
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if applicable
- [x] My changes generate no new warnings
- [x] I have added tests that prove my code is effective or that my implementation works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backend/__tests__/unit/controllers/userAuth.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    //     "msg": "Login successful"
+    //   }
+    //
+    //   return login(req, res).then(data => {
+    //     expect(data.json.status).toBe(expectedResponse.status);
+    //     expect(data.json.success).toBe(expectedResponse.success);
+    //     expect(data.json.msg).toBe(expectedResponse.msg);
+    //   });
+    //
+    // });
```
Comment (tysonkaufmann) [inline backend/__tests__/unit/controllers/userAuth.js]: Is this supposed to be commented out? Or will this test be implemented in the future?

**Review by tysonkaufmann — COMMENTED** [review]
Looks great to me - only question I have is the one test that is commented out. Is it a placeholder for now or should it be removed?

File: backend/__tests__/unit/controllers/userAuth.js (hunk 1 above)
Comment (Mitul264) [reply backend/__tests__/unit/controllers/userAuth.js]: Should be removed as we have done successful login code coverage in the integration test

**Review by tysonkaufmann — APPROVED** [review]
Great job!

**Review by chucklee9693 — APPROVED** [review]
Looks good to me! Great work!

**Pull request merged**